### ISSUE TITLE
Investigate high execution payload verification time

### ISSUE THREAD

**Issue by twoeths** [ChainSafe/lodestar#5429]
**Describe the bug**

Sometimes it takes significant time to verify execution engine payload on test mainnet nodes

<img width="1309" alt="Screenshot 2023-04-28 at 08 41 45" src="https://user-images.githubusercontent.com/10568965/235034066-4cc06ca7-ba56-4582-a671-3e51d234ad05.png">

Some logs:

```
1143794:Apr-26 23:46:31.425[chain]         verbose: Verified execution payload slot=6310730, recvToVerifiedExecPayload=6.816999912261963
Apr-26 23:48:05.877[chain]         verbose: Verified execution payload slot=6310738, recvToVerifiedExecPayload=5.330000162124634
```
it's likely lodestar is busy at that time, not the execution layer's issue

**Expected behavior**
- <0.5s consistently

**Comment by twoeths** [comment]
was able to take a profile of this block

```
Apr-28 02:37:16.797[chain]         verbose: Verified execution payload slot=6318784, recvToVerifiedExecPayload=1.9059998989105225
```

full log:
```
beacon-2023-04-29.log:292138:Apr-29 05:07:50.846[chain]         verbose: Verified execution payload slot=6326737, recvToVerifiedExecPayload=1.9059998989105225
devops@feat1-mainnet-hzax41:~/beacon$ grep -e "6326737" -rn beacon-2023-04-29.log
292041:Apr-29 05:07:43.367[chain]         verbose: Running prepareForNextSlot nextEpoch=197711, prepareSlot=6326737, headSlot=6326736, headRoot=0xadaca0a5724a5ccca2d7cbd56277c2ab7a5e24d1ec510bc54a8710a6f7166902, isEpochTransition=false
292108:Apr-29 05:07:47.000[chain]         verbose: Clock slot slot=6326737
292129:Apr-29 05:07:48.961[network]       verbose: Received gossip block slot=6326737, root=0x6510…54a5, curentSlot=6326737, peerId=16Uiu2HAmVCyDPHdUYYv9mMY2Z2DDCm2WVRd1PBK5nc4i4F5y6P4K, delaySec=1.940000057220459, recvToVal=0.020999908447265625
292130:Apr-29 05:07:49.301[chain]         verbose: Transitioned gossip block slot=6326737, recvToTransition=0.36100006103515625
292131:Apr-29 05:07:49.336[chain]         verbose: Verified block signatures slot=6326737, recvToSigVer=0.3959999084472656
292138:Apr-29 05:07:50.846[chain]         verbose: Verified execution payload slot=6326737, recvToVerifiedExecPayload=1.9059998989105225
292142:Apr-29 05:07:50.933[chain]           debug: Persisted block to hot DB slot=6326737, root=0x651026bba0f87a45e2efa95534f75c7c0aaa6b8d4f196d438529461a5f4954a5
292143:Apr-29 05:07:50.934[chain]         verbose: Added block to forkchoice and state cache slot=6326737, root=0x651026bba0f87a45e2efa95534f75c7c0aaa6b8d4f196d438529461a5f4954a5
292144:Apr-29 05:07:51.149[chain]         verbose: New chain head slot=6326737, root=0x651026bba0f87a45e2efa95534f75c7c0aaa6b8d4f196d438529461a5f4954a5, delaySec=4.148999929428101
292145:Apr-29 05:07:51.169[chain]         verbose: Imported block slot=6326737, recvToImportedBlock=2.2289998531341553
292146:Apr-29 05:07:51.169[chain]         verbose: Block processed slot=6326737, root=0x651026bba0f87a45e2efa95534f75c7c0aaa6b8d4f196d438529461a5f4954a5, delaySec=4.168999910354614
292149:Apr-29 05:07:53.279[]                 info: Synced - slot: 6326737 - head: 0x6510…54a5 - exec-block: valid(17149575 0x12e8…) - finalized: 0x6793…4739:197708 - peers: 55
292209:Apr-29 05:07:55.175[chain]         verbose: Running prepareForNextSlot nextEpoch=197711, prepareSlot=6326738, headSlot=6326737, headRoot=0x651026bba0f87a45e2efa95534f75c7c0aaa6b8d4f196d438529461a5f4954a5, isEpochTransition=false
```


[0428_mainnet_node_IO_lag.cpuprofile.zip](https://github.com/ChainSafe/lodestar/files/11349676/0428_mainnet_node_IO_lag.cpuprofile.zip)

1. at 32490ms, start `verifyBlocksExecutionPayload`
2. at 32500, it runs `verifyBlocksStateTransitionOnly` and finish in around 32700
3. at 34377ms, process response from `verifyBlocksExecutionPayload`
4. at 34378ms, it adds block to db
5. at 37396ms, it processes state to add checkpoint balance cache (just next command to (note that `this.checkpointBalancesCache.processState` is just next to `this.db.block.add` in our code)

There are a lot of time between the gaps (2 to 3, 4 to 5) it just do network I/O tasks:
- handleReceivedRpc
- handleReceivedMessage
- chacha20Poly1305Decrypt
![Screenshot 2023-04-28 at 11 15 49](https://user-images.githubusercontent.com/10568965/235052788-f81f35c3-526e-4c96-9788-88004a17ca72.png)


==> the I/O lag seems to be not at lodestar side but at lower network stack libraries (that's why it happens in v1.7.2 too)

**Comment by twoeths** [comment]
we can't control `chacha20Poly1305Decrypt` in lodestar, however we can control `handleReceivedMessage` like in #5413: if we get a gossip block, we can put `handleReceivedMessage` in a lodestar queue to control concurrent messages at the same and yield more to event loop, until the block is processed

I have a branch to always put `handleReceivedMessage` in lodestar queue, it make lodestar has low number of subnet mesh peers because we don't process them fast enough, we only want to do it in a short window within the first 1/3 of slot

**Comment by twoeths** [comment]
resolved with v1.10.0

<img width="1311" alt="Screenshot 2023-08-20 at 13 07 56" src="https://github.com/ChainSafe/lodestar/assets/10568965/50715878-efed-48b3-a597-b0e465530c40">
